modules/api/src/main/java/org/apache/ignite/compute/JobTarget.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.compute;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+import org.apache.ignite.network.ClusterNode;
+import org.apache.ignite.table.Tuple;
+import org.apache.ignite.table.mapper.Mapper;
+
+/**
+ * Job execution target.
+ */
+public interface JobTarget {
+    /**
+     * Creates a job target for a specific node.
+     *
+     * @param node Node.
+     * @return Job target.
+     */
+    static JobTarget node(ClusterNode node) {
+        return new NodesJobTarget(Set.of(node));
+    }
+
+    /**
+     * Creates a job target for any node from the provided collection.
+     *
+     * @param nodes Collection of nodes.
+     * @return Job target.
+     */
+    static JobTarget anyNode(ClusterNode... nodes) {
+        return new NodesJobTarget(Set.of(nodes));
+    }
+
+    /**
+     * Creates a job target for any node from the provided collection.
+     *
+     * @param nodes Collection of nodes.
+     * @return Job target.
+     */
+    static JobTarget anyNode(Collection<ClusterNode> nodes) {
+        return new NodesJobTarget(new HashSet<>(nodes));
+    }
```

Comment:
Should we mention that the collection of nodes is treated as a Set (duplicates are ignored)?

Reply:
I don't think this matters with "any node" semantics because we don't explain how a node is picked from the provided collection. There is no guarantee of fairness or randomness on the API level.